File: CodeLibrary/TabHost/BottomNavigationView/AndroidJavaDesignBottomNavigationView/app/src/main/java/com/jackshi/androidjavatabhostfordesign/utils/MyPagerAdapter.java
package com.jackshi.androidjavatabhostfordesign.utils;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.view.MenuItem;

import java.util.ArrayList;

public class MyPagerAdapter extends FragmentPagerAdapter {
    ArrayList<Fragment> mList;
    MenuItem mMenuItem;
    public MyPagerAdapter(FragmentManager fm, ArrayList<Fragment> list, MenuItem menuItem) {
        super(fm);
        this.mList=list;
        this.mMenuItem=menuItem;
    }

    @Override
    public int getCount() {
        return mList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mMenuItem.getTitle();
    }

    @Override
    public Fragment getItem(int position) {
        return mList.get(position);
    }
}

File: CodeLibrary/TabHost/BottomNavigationView/AndroidXkotlinBottomNavigationView/app/src/main/java/com/jackshi/androidxkotlinbottomnavigationview/utils/MyPagerAdapter.kt
package com.jackshi.androidxkotlinbottomnavigationview.utils

import android.view.MenuItem
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MyPagerAdapter(
    fm: FragmentManager,
    list: ArrayList<Fragment>,
    menuItem: MenuItem
) : FragmentPagerAdapter(fm) {
    private var mList: ArrayList<Fragment> = ArrayList()
    private var mMenuItem: MenuItem

    init {
        mList = list
        mMenuItem = menuItem
    }

    override fun getPageTitle(position: Int): CharSequence {
        return mMenuItem.title
    }

    override fun getItem(position: Int): Fragment {
        return mList[position]
    }

    override fun getCount(): Int {
        return mList.size
    }
}